project_task_id_in_display_name/views/project_task_views.xml

```diff
@@ -0,0 +1,77 @@
+            </field>
+            <field name="name" position="attributes">
+                <attribute name="class">oe_edit_only</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_task_kanban" model="ir.ui.view">
+        <field name="name">Task Kanban With Display Name Instead Of Name</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_kanban"/>
+        <field name="arch" type="xml">
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/s/field[@name='name']" position="before">
+                <field name="display_name"/>
+            </xpath>
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/s/field[@name='name']" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath>
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/t/field[@name='name']" position="before">
+                <field name="display_name"/>
+            </xpath>
+            <xpath expr="//strong[hasclass('o_kanban_record_title')]/t/field[@name='name']" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_task_tree2" model="ir.ui.view">
+        <field name="name">Task List: Display Name Instead Of Name</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_tree2"/>
+        <field name="arch" type="xml">
+            <field name="name" position="before">
+                <field name="display_name"/>
+            </field>
+            <field name="name" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="project_task_view_tree_activity" model="ir.ui.view">
+        <field name="name">Task Next Activities List: Display Name Instead Of Name</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.project_task_view_tree_activity"/>
+        <field name="arch" type="xml">
+            <field name="name" position="before">
+                <field name="display_name"/>
+            </field>
+            <field name="name" position="attributes">
+                <attribute name="invisible">1</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_task_search_form" model="ir.ui.view">
+        <field name="name">Task Search With ID Searched In Name</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_search_form"/>
+        <field name="arch" type="xml">
+            <field name="name" position="attributes">
+                <attribute name="filter_domain">['|', ('id_string', '=', self), ('name', 'ilike', self)]</attribute>
```

Comment:
Hello @rivo2302 
We dont need to add search by id_string, in vanilla module we can search by ID
[See Link](https://github.com/odoo/odoo/blob/16.0/addons/project/views/project_views.xml#L19)